src/main/java/ch/njol/skript/util/Date.java

```diff
@@ -1,142 +1,140 @@
-	}
-	
+
 	/**
-	 * Get the timestamp of this date
+	 * Creates a new Date with the provided timestamp and timezone.
 	 *
-	 * @return The timestamp in milliseconds
+	 * @param timestamp The timestamp in milliseconds.
+	 * @param zone The timezone to use.
 	 */
-	public long getTimestamp() {
-		return timestamp;
+	public Date(long timestamp, TimeZone zone) {
+		super(timestamp - zone.getOffset(timestamp));
 	}
-	
+
 	/**
 	 * Add a {@link Timespan} to this date
 	 *
-	 * @param span Timespan to add
+	 * @param other Timespan to add
 	 */
-	public void add(final Timespan span) {
-		timestamp += span.getMilliSeconds();
+	public void add(Timespan other) {
+		timestamp += other.getAs(Timespan.TimePeriod.MILLISECOND);
+		setTime(timestamp);
 	}
-	
+
 	/**
 	 * Subtract a {@link Timespan} from this date
 	 *
-	 * @param span Timespan to subtract
+	 * @param other Timespan to subtract
 	 */
-	public void subtract(final Timespan span) {
-		timestamp -= span.getMilliSeconds();
+	public void subtract(Timespan other) {
+		timestamp -= other.getAs(Timespan.TimePeriod.MILLISECOND);
+		setTime(timestamp);
 	}
-	
+
+	/**
+	 * Returns the difference between this date and another date as a {@link Timespan}.
+	 *
+	 * @param other The other date.
+	 * @return The difference between the provided dates as a {@link Timespan}.
+	 */
+	public Timespan difference(Date other) {
+		return new Timespan(Math.abs(timestamp - other.timestamp));
+	}
+
 	/**
 	 * Get a new instance of this Date with the added timespan
 	 *
-	 * @param span Timespan to add to this Date
+	 * @param other Timespan to add to this Date
 	 * @return New Date with the added timespan
 	 */
-	public Date plus(Timespan span) {
-		return new Date(timestamp + span.getMilliSeconds());
+	public Date plus(Timespan other) {
+		return new Date(timestamp + other.getAs(Timespan.TimePeriod.MILLISECOND));
 	}
 	
 	/**
 	 * Get a new instance of this Date with the subtracted timespan
 	 *
-	 * @param span Timespan to subtract from this Date
+	 * @param other Timespan to subtract from this Date
 	 * @return New Date with the subtracted timespan
 	 */
-	public Date minus(Timespan span) {
-		return new Date(timestamp - span.getMilliSeconds());
+	public Date minus(Timespan other) {
+		return new Date(timestamp - other.getAs(Timespan.TimePeriod.MILLISECOND));
 	}
-	
+
+	/**
+	 * @deprecated Use {@link #getTime()} instead.
+	 */
+	@Deprecated(forRemoval = true)
+	public long getTimestamp() {
+		return timestamp;
+	}
+
 	@Override
 	public int hashCode() {
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + (int) (timestamp ^ (timestamp >>> 32));
-		return result;
+		return 31 + Long.hashCode(timestamp);
 	}
-	
+
 	@Override
-	public boolean equals(final @Nullable Object obj) {
+	public boolean equals(@Nullable Object obj) {
 		if (this == obj)
 			return true;
 		if (obj == null)
 			return false;
-		if (!(obj instanceof Date))
+		if (!(obj instanceof Date other))
```

Comment:
Do you want to allow equality comparison with Java dates? If so you might wish to compare with j.u.Date's time instead. (Or you may be able to use j.u.Date's equality check as-is without overriding.)

Reply:
oops forgot to update it